### ISSUE TITLE
Fontset configuration messed up

### ISSUE THREAD

**Issue by guraltsev** [doomemacs/doomemacs#3296]
**What did you expect to happen?**
I expect cyrillic and greek letters to be displayed in `doom-font` by default. 
It is not reasonable for iconic fonts to be used to display normal non-latin letter character ranges.

**What actually happened?**
Cyrillic  and greek letters and many more are display using the iconic font `FontAwesome`

**Steps to reproduce:**
Run doom using default template `config.el`, `init.el` and `package.el`
Now open any buffer (scratch) and insert
```
;; Привет
```
Position the cursor on the string and run 
`M-x describe-char` and note the wrong font being used. 

You may also want to try by including 
``` (setq doom-font (font-spec :family "Firacode NF" :size 18)) ```
into your `config.el` file. Nothing changes anyway.

**System information:**
<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS LCMS2
        build      Dec 25, 2019
        buildopts  (--without-compress-install --without-dbus --with-modules 'CFLAGS=-O2 -g3')
        windowsys  w32
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD
        dir        ~/.config/doom/
system  type       windows-nt
        config     x86_64-w64-mingw32
        shell      C:/bin/emax64-26.3/libexec/emacs/26.3/x86_64-w64-mingw32/cmdproxy.exe
        uname      n/a
        path       (c:/bin/ C:/Program Files (x86)/Common Files/Oracle/Java/javapath C:/Windows/system32 C:/Windows C:/Windows/System32/Wbem C:/Windows/System32/WindowsPowerShell/v1.0/ C:/Windows/System32/OpenSSH/ C:/ProgramData/chocolatey/bin C:/Program Files/Calibre2/ C:/Program Files (x86)/gnupg/bin C:/Program Files/TortoiseGit/bin C:/Program Files (x86)/ZeroTier/One/ C:/Program Files (x86)/NVIDIA Corporation/PhysX/Common C:/Program Files (x86)/Intel/Intel(R) Management Engine Components/DAL C:/Program Files/Intel/Intel(R) Management Engine Components/DAL C:/Program Files/ImageMagick-7.0.10-Q16 C:/texlive/2020/bin/win32 C:/Program Files (x86)/Wolfram Research/WolframScript/ ~/AppData/Local/Microsoft/WindowsApps C:/tools/Cmder C:/Program Files/Oracle/VirtualBox ~/AppData/Local/Pandoc/ . c:/bin/emax64-26.3/libexec/emacs/26.3/x86_64-w64-mingw32)
config  envfile    nil
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline ophints (popup +defaults) vc-gutter vi-tilde-fringe workspaces :editor file-templates fold snippets :emacs dired electric undo vc :checkers syntax :tools (eval +overlay) lookup magit :lang emacs-lisp markdown org sh :config (default +bindings +smartparens))
        packages   (n/a)
        unpin      (n/a)
        elpa       (n/a)
```
</details>

**Comment by guraltsev** [comment]
By the way, this my be related to [](https://github.com/hlissner/doom-emacs/issues/3250)

**Comment by krvital** [comment]
Hi! I have the same problem and it annoys me. I would like to write org documents in Russian sometimes but can't stand the sight of the difference between two fonts on one page.

Maybe you know any workaround for this? 

I've tried to debug this and noticed that it works (by this I mean it properly loads font for Cyrillic letters) when I intentionally break doom-theme loading with the next line 
`(setq doom-theme 'defualt)`

**Comment by licht1stein** [comment]
I recently installed Doom Emacs and noticed the same problem. So I used the Chinese font solution @guraltsev mention to make this stupid workaround:

```elisp
(dolist (charset '(koi8))
  (set-fontset-font (frame-parameter nil 'font)
    charset (font-spec :family "Fira Code" :height 90)))
```

**Comment by George9000** [comment]
Ran into this with unicode characters in Julia. I already had in `config.el`

```
(setq doom-unicode-font (font-spec :family "JuliaMono" :size 16 :weight 'medium))
``` 
Above did not work till I added

```
(dolist (charset '(unicode))
  (set-fontset-font (frame-parameter nil 'font)
    charset (font-spec :family "JuliaMono" :size 16 :weight 'medium)))
```

_Both_ configs were needed. Neither one alone was enough. This [post](https://www.reddit.com/r/DoomEmacs/comments/v0voho/unicode_characters_not_displayed_properly_font/iak8i0d/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) was helpful.

**Comment by George9000** [comment]
@hpfr Thank you for attempting a fix. I tried a fresh install by cloning your fork with the latest commit. This did not fix my issue. In fact, it seemed to cause a _regression_ as neither just the `setq doom-unicode-font` nor adding the workaround posted above successfully changed the unicode font to the heavier weight, larger size. I kept getting a spindly, small font size. To get the larger effect, as in the picture below, I had to go back to the doom `master` branch and use the workaround above.

![julia_mono_example](https://github.com/doomemacs/doomemacs/assets/16549651/304aa668-2910-4177-9f2c-73f22c6d5099)
